public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts

```diff
@@ -60,7 +64,9 @@ export const updateMaxDataPoints = createAction<{ refId: string; maxDataPoints:
 export const updateMinInterval = createAction<{ refId: string; minInterval: string }>('updateMinInterval');
 
 export const resetToSimpleCondition = createAction('resetToSimpleCondition');
-
+export const optimizeReducer = createAction<{ updatedQueries: AlertQuery[]; expressionQueries: AlertQuery[] }>(
```

Comment:
Can we rename it to something like `optimizeReduceExpression`? `optimizeReducer` looks like a redux reducer